modules/mlflow/mlflow-fargate/stack.py

```diff
@@ -0,0 +1,111 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, List, Optional, cast
+
+from aws_cdk import Duration, Stack, Tags
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from aws_cdk import aws_ecs as ecs
+from aws_cdk import aws_ecs_patterns as ecs_patterns
+from aws_cdk import aws_iam as iam
+from constructs import Construct, IConstruct
+
+
+class MlflowFargateStack(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        app_prefix: str,
+        vpc_id: str,
+        subnet_ids: List[str],
+        ecs_cluster_name: Optional[str],
+        service_name: Optional[str],
+        ecr_repo_name: str,
+        task_cpu_units: int,
+        task_memory_limit_mb: int,
+        artifacts_bucket_name: str,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=app_prefix[:64])
+
+        role = iam.Role(
+            self,
+            "TaskRole",
+            assumed_by=iam.ServicePrincipal(service="ecs-tasks.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonS3FullAccess"),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonECS_FullAccess"),
+            ],
+        )
+
+        vpc = ec2.Vpc.from_lookup(self, "vpc", vpc_id=vpc_id)
+        subnets = [ec2.Subnet.from_subnet_id(self, f"sub-{subnet_id}", subnet_id) for subnet_id in subnet_ids]
+
+        cluster = ecs.Cluster(
+            self,
+            "EcsCluster",
+            cluster_name=ecs_cluster_name,
+            vpc=vpc,
+        )
+        self.cluster = cluster
+
+        task_definition = ecs.FargateTaskDefinition(
+            self,
+            "MlflowTask",
+            task_role=role,
+            cpu=task_cpu_units,
+            memory_limit_mib=task_memory_limit_mb,
+        )
+
+        container = task_definition.add_container(
+            "ContainerDef",
+            # TODO: add ability to pull specific tag
+            image=ecs.ContainerImage.from_ecr_repository(
+                repository=ecr.Repository.from_repository_name(
+                    self,
+                    "ECRRepo",
+                    repository_name=ecr_repo_name,
+                ),
+            ),
+            environment={
+                "BUCKET": f"s3://{artifacts_bucket_name}",
+                # TODO: Add persistence
+                # "HOST": database.db_instance_endpoint_address,
```

Comment:
Will this be configurable (i.e. add in RDS persistence if configured)

Reply:
Yep. Also in case user wants to use the default file store for persistence we probably need an EBS volume -- checking this.

Reply:
Added EFS by default